### ISSUE TITLE
Create URL Params based filter so user can bookmark the search history

### ISSUE THREAD

**Issue by vamsi981** [SharePoint/sp-dev-solutions#57]
Hi, how to enable URL params so that users can bookmark the search URL and share it across. is there any guidance for that?

**Comment by YannickRe** [comment]
Hi @vamsi981 

If I understand correctly what you are trying to achieve, I'm afraid that's currently not possible.
You can put a Search Box web part on a page that "reads" from the Page Environment, Query String and URL Fragment and then passes it on to the Search Results web part. If a user changes the query in the Search Box web part, it doesn't update the URL Fragment at the moment.

@wobba @FranckyC Might be a quick enhancement, using history.pushState? Do we need a toggle for this int he property panel of the Search Box web part?

**Comment by wobba** [comment]
Shouldn't break anything by switching behavior I think(?) so a toggle seems unne.

**Comment by YannickRe** [comment]
No, but it would add the search query as a hash for everyone without an option to have the current behavior.

**Comment by FranckyC** [comment]
It could work of course but I think it could also lead to infinite loops or unnecessary calls with wrong WP configuration. Maybe a manual option like 'Save this query as bookmark" in the SB component can avoid this.

**Comment by vamsi981** [comment]
@YannickRe @FranckyC @wobba  is it possible to get search as bellow [ReactiveSearch](https://opensource.appbase.io/reactivesearch/demos/ecommerce/) with URL params for filters and search text so that user can bookmark or share the output to other instead of others searching from start.

**Comment by wobba** [comment]
@vamsi981 anything is possible if it's implemented correct :) If you have a searchbox and pick up a query param for it, that can indeed be shared.

**Comment by YannickRe** [comment]
It would mean making sure that the querystring/url params are parsed and applied upon entering the page + adjustments to SearchBox, SearchRefiners and SearchPagination to add changes to the url (making sure that we don't introduce loops or break existing behavior).

Not something you create on a lazy Sunday afternoon 🙂.

**Comment by wobba** [comment]
@YannickRe could have a special param: `?startquery=foo` which is only read upon first load - or, always use `#query=`, which worked fine with 2013  `searchpage.aspx#k=something`. But it has to work with the current architecture.
